integr8ly/keycloak-operator
# Intly 1325

## Motivation

Allow unmanaged (create only) realms. Those realms won't be reconciled by the operator and can be modified by the user.

## What

Adds a (optional) `createOnly` flag to the KeycloakRealm. If set to true the realm will be created but not reconciled.

## Verification Steps

1. Deploy the keycloak operator from this branch into an Integr8ly installation
1. Go to the sso namespace and edit the `openshift` KeycloakRealm (Resources -> Other resources)
1. In the spec section add `createOnly: true`
1. Now log in to keycloak, make sure the openshift realm is selected and create a new identity provider (e.g. Github)
1. Wait a few minutes and make sure the provider is not deleted
1. Now go back to the KeycloakRealm resource and set `createOnly` to false
1. The realm will be reconciled again and your provider should get deleted

## Progress
